test/dotnet.Tests/dotnet-msbuild/GivenDotnetRunInvocation.cs

```diff
@@ -8,28 +8,36 @@ namespace Microsoft.DotNet.Cli.MSBuild.Tests
     {
-        private static readonly string WorkingDirectory =
-            TestPathUtilities.FormatAbsolutePath(nameof(GivenDotnetRunInvocation));
+        private string WorkingDirectory { get; init; }
+
+        public GivenDotnetRunInvocation(ITestOutputHelper log)
+        {
+            var tam = new TestAssetsManager(log);
+            WorkingDirectory = tam.CopyTestAsset("HelloWorld").WithSource().Path;
+        }
 
         [Theory]
-        [InlineData(new string[] { "-p:prop1=true" }, new string[] { "-p:prop1=true" })]
-        [InlineData(new string[] { "--property:prop1=true" }, new string[] { "-p:prop1=true" })]
-        [InlineData(new string[] { "--property", "prop1=true" }, new string[] { "-p:prop1=true" })]
-        [InlineData(new string[] { "-p", "prop1=true" }, new string[] { "-p:prop1=true" })]
-        [InlineData(new string[] { "-p", "prop1=true", "-p", "prop2=false" }, new string[] { "-p:prop1=true", "-p:prop2=false" })]
-        [InlineData(new string[] { "-p:prop1=true;prop2=false" }, new string[] { "-p:prop1=true;prop2=false" })]
-        [InlineData(new string[] { "-p", "MyProject.csproj", "-p:prop1=true" }, new string[] { "-p:prop1=true" })]
+        [InlineData(new string[] { "-p:prop1=true" }, new string[] { "--property:prop1=true" })]
+        [InlineData(new string[] { "--property:prop1=true" }, new string[] { "--property:prop1=true" })]
+        [InlineData(new string[] { "--property", "prop1=true" }, new string[] { "--property:prop1=true" })]
+        [InlineData(new string[] { "-p", "prop1=true" }, new string[] { "--property:prop1=true" })]
+        [InlineData(new string[] { "-p", "prop1=true", "-p", "prop2=false" }, new string[] { "--property:prop1=true", "--property:prop2=false" })]
+        [InlineData(new string[] { "-p:prop1=true;prop2=false" }, new string[] { "--property:prop1=true", "--property:prop2=false" })]
+        [InlineData(new string[] { "-p", "MyProject.csproj", "-p:prop1=true" }, new string[] { "--property:prop1=true" })]
         // The longhand --property option should never be treated as a project
-        [InlineData(new string[] { "--property", "MyProject.csproj", "-p:prop1=true" }, new string[] { "-p:MyProject.csproj", "-p:prop1=true" })]
-        [InlineData(new string[] { "--disable-build-servers" }, new string[] { "-p:UseRazorBuildServer=false", "-p:UseSharedCompilation=false", "/nodeReuse:false" })]
+        [InlineData(new string[] { "--property", "MyProject.csproj", "-p:prop1=true" }, new string[] { "--property:MyProject.csproj", "--property:prop1=true" })]
+        [InlineData(new string[] { "--disable-build-servers" }, new string[] { "--property:UseRazorBuildServer=false", "--property:UseSharedCompilation=false", "/nodeReuse:false" })]
         public void MsbuildInvocationIsCorrect(string[] args, string[] expectedArgs)
         {
+
+            string[] constantRestoreArgs = ["-nologo", "-verbosity:quiet"];
```

Comment:
Since we're unifying args better now we get these constants that we have to be aware of.